bin/watch-packages-and-rebuild.js

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env node
+const fs = require( 'fs' );
+const path = require( 'path' );
+const { spawnSync } = require( 'child_process' );
+const chokidar = require( 'chokidar' );
+
+const packagesDirectoryPath = path.join( '.', 'packages' );
+const watcher = chokidar.watch( packagesDirectoryPath, {
+	ignored: /^\./,
+	persistent: true,
+	ignoreInitial: true,
+} );
+watcher.on( 'change', handleChange ).on( 'add', handleChange ).on( 'unlink', handleChange );
+
+let rebuildQueue = [];
+
+setTimeout( processRebuildQueue, 100 );
+
+function processRebuildQueue() {
+	rebuildQueue.map( rebuildPackage );
+	rebuildQueue = [];
+	setTimeout( processRebuildQueue, 100 );
+}
+
+function handleChange( filePath ) {
+	if ( isPathIgnored( filePath ) ) {
+		return;
+	}
+	console.log( 'heard change in', filePath );
+	const packageName = getPackageNameFromFilePath( filePath );
+	if ( packageName && ! rebuildQueue.includes( packageName ) ) {
+		rebuildQueue.push( packageName );
+	}
+}
+
+function isPathIgnored( filePath ) {
+	const filePathPieces = filePath.split( path.sep );
+	if ( filePathPieces.length < 2 ) {
+		return true;
+	}
+	const ignoredPaths = [ 'dist', 'test', 'tests' ];
+	if ( ignoredPaths.find( ( ignored ) => filePathPieces.includes( ignored ) ) ) {
+		return true;
+	}
+	const ignoredExtensions = [ '.css', '.md', '.d.ts' ];
+	if ( ignoredExtensions.find( ( ignored ) => filePath.endsWith( ignored ) ) ) {
+		return true;
+	}
+	return false;
+}
+
+function getPackageNameFromFilePath( filePath ) {
```

Comment:
This looks like a lot of extra work to get the package name, only to pass it to `yarn worspace`.

In theory you don't need the package name, only the package dir. you could do something like `spawSync ('yarn', ['prepare'], { cwd: <package-dir> })`

Reply:
Smart idea. Wouldn't occor to me.